lib/chef/resource/rhsm_subscription.rb

```diff
@@ -32,11 +32,11 @@ class RhsmSubscription < Chef::Resource
         name_property: true
 
       action :attach, description: "Attach the node to a subscription pool." do
-        execute "Attach subscription pool #{new_resource.pool_id}" do
-          command "subscription-manager attach --pool=#{new_resource.pool_id}"
-          default_env true
-          action :run
-          not_if { subscription_attached?(new_resource.pool_id) }
+        unless subscription_attached?(new_resource.pool_id)
+          converge_by("attach subscription pool #{new_resource.pool_id}") do
+            shell_out!("subscription-manager attach --pool=#{new_resource.pool_id}")
+            build_resource(:package, "rhsm_subscription-#{new_resource.pool_id}-flush_cache").run_action(:flush_cache)
```

Comment:
@lamont-granquist  Will this also run the default action (i.e. `:install`)? Or is this all good?

Reply:
In other words, do I need to do this?

```ruby
            build_resource(:package, "rhsm_subscription-#{new_resource.pool_id}-flush_cache") do
              action :nothing
            end.run_action(:flush_cache
```

Reply:
no you're not using `declare_resource` so its not being put on the resource_collection and won't be converged by the normal process.  only the actions you run get run.

`build_resource` is just some magic around getting a `new` instance of the right package resource.

This also means that it won't cause the resource you are running to be updated when it updates -- which is why you shouldn't generally ever use build_resource and should at least use declare_resource -- but here we're doing it deliberately because we're flushing the cache and that has nothing to do with the underlying state of the machine changing.  Arguably using a resource for this is philosophically wrong, but its the public API that we have.